tests/testthat/test-namespace.R

```diff
@@ -0,0 +1,8 @@
+test_that("is_s3_generic", {
+  func <- function(x) {
```

Comment:
could you add a test case for the error described in #1161, namely for a function where the last expression is namespace-qualified?